src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/ActionPlanEntity.kt

```diff
@@ -63,8 +71,8 @@ class ActionPlanEntity(
     /**
      * Returns new un-persisted [ActionPlanEntity] for the specified prisoner with an empty collection of [GoalEntity]s
      */
-    fun newActionPlanForPrisoner(prisonNumber: String): ActionPlanEntity =
-      ActionPlanEntity(prisonNumber = prisonNumber, goals = mutableListOf())
+    fun newActionPlanForPrisoner(prisonNumber: String, reviewDate: LocalDate?): ActionPlanEntity =
+      ActionPlanEntity(reference = UUID.randomUUID(), prisonNumber = prisonNumber, reviewDate = reviewDate, goals = mutableListOf())
```

Comment:
Slightly uncomfortable with this in that we have a persistence concern effectively deciding how to create a domain object .... 🤔 

IE. the reference and review date of an ActionPlan are actually domain concerns I think (though I know our mappers that map from the REST API to Domain objects for Goals and Steps do this currently .... 🤔 )

The purist in me is nagging at me! 🤣 - if we changed our DB layer to mongo or dynamo or something non-JPA (not that we would) , we would need to move/re-implement that logic .... feels wrong ... 🤔 

Lets go with it for now, but it is eating me alive! 🤣